controller/search_blackbox_test.go

```diff
@@ -1068,6 +1068,62 @@ func (s *searchControllerTestSuite) TestSearchByJoinedData() {
 	})
 }
 
+// TestOrderOfExecution tests order of execution of workitems with tree-view and without tree-view filter
+// Suppose we have this topology:
+// Parent1
+// 	|_ Child1
+// Parent2
+//	|_ Child2
+//	|_ Child3
+// Parent3
+// Parent4
+// Parent5
+func (s *searchControllerTestSuite) TestOrderOfExecution() {
+	resetFn := s.DisableGormCallbacks()
+	defer resetFn()
+
+	fxt := tf.NewTestFixture(s.T(), s.DB,
+		tf.WorkItems(8, tf.SetWorkItemTitles("Parent1", "Parent2", "Parent3", "Parent4", "Parent5", "Child1", "Child2", "Child3")),
+		tf.WorkItemLinksCustom(3, func(fxt *tf.TestFixture, idx int) error {
+			l := fxt.WorkItemLinks[idx]
+			l.LinkTypeID = link.SystemWorkItemLinkTypeParentChildID
+			switch idx {
+			case 0:
+				l.SourceID = fxt.WorkItemByTitle("Parent1").ID
+				l.TargetID = fxt.WorkItemByTitle("Child1").ID
+			case 1:
+				l.SourceID = fxt.WorkItemByTitle("Parent2").ID
+				l.TargetID = fxt.WorkItemByTitle("Child2").ID
+			case 2:
+				l.SourceID = fxt.WorkItemByTitle("Parent2").ID
+				l.TargetID = fxt.WorkItemByTitle("Child3").ID
+			}
+			return nil
+		}),
+	)
+	spaceIDStr := fxt.Spaces[0].ID.String()
+
+	// Without Tree View
+	filter := fmt.Sprintf(`{"space": "%[1]s"}`, spaceIDStr)
+	_, result := test.ShowSearchOK(s.T(), nil, nil, s.controller, &filter, nil, nil, nil, nil, &spaceIDStr)
+	// then
+	require.NotEmpty(s.T(), result.Data)
+	for i, wi := range result.Data {
+		i = len(result.Data) - 1 - i
+		assert.Equal(s.T(), fxt.WorkItems[i].Fields[workitem.SystemOrder], wi.Attributes[workitem.SystemOrder])
+	}
+
+	// With Tree View
+	filter = fmt.Sprintf(`{"space": "%[1]s", "$OPTS":{"%[2]s": true}}`, spaceIDStr, search.OptTreeViewKey)
+	_, result = test.ShowSearchOK(s.T(), nil, nil, s.controller, &filter, nil, nil, nil, nil, &spaceIDStr)
+	// then
+	require.NotEmpty(s.T(), result.Data)
+	for i, wi := range result.Data {
+		i = len(result.Data) - 1 - i
```

Comment:
While it might seem nice to reuse the logic from the test before, I suggest to check for how many items where returned as well. The number should differ between with/without tree view, correct? I would rather have 5 or more checks that test for equality with explicit indexing or naming (e.g. by `fxt.WorkItemByTitle("Parent1")`) than having a for loop.

Reply:
`The number should differ between with/without tree view, correct?`

No. result.Data has same workitems in both cases.

` I would rather have 5 or more checks that test for equality with explicit indexing or naming (e.g. by fxt.WorkItemByTitle("Parent1")) than having a for loop.`

Why?

I can see you use for loops to test for equality instead of testing with explicit indexing -> https://github.com/fabric8-services/fabric8-wit/pull/1842/files#diff-a2bd8cd90ce50fc5463d8b3539620c92R1087

Reply:
> Why?
> 
> I can see you use for loops to test for equality instead of testing with explicit indexing -> https://github.com/fabric8-services/fabric8-wit/pull/1842/files#diff-a2bd8cd90ce50fc5463d8b3539620c92R1087

@DhritiShikhar I'm using a loop to address dynamic data. You're is fixed.

Reply:
> The number should differ between with/without tree view, correct?
> 
> No. result.Data has same workitems in both cases.

Oh, yes. Correct.